src/main/java/org/apache/sling/resourceresolver/impl/mapping/MapEntries.java

```diff
@@ -127,7 +128,9 @@
 
     private Map <String,List <String>> vanityTargets;
 
-    private Map<String, Map<String, String>> aliasMap;
+    private volatile Map<String, Map<String, String>> aliasMap;
```

Comment:
Two volatile flags don't ensure consistency. Either use a synchronized block and non-volatile fileds or ensure thread safety in some other manner.

Reply:
@rombert  thx - in fact the volatile is obsolete and was just a leftover of an attempt to fix an issue that turned out to be caused by the mocking flow - the map used already is a ConcurrentHashMap - so I'll drop the volatile marker